### PR TITLE
fix: add default value for useSize

### LINKED ISSUE #2043: useSize
useSize return undefined at first render 

"react": "18.2.0",
 "ahooks": "^3.7.4",

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/alibaba/hooks/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

- close #2043

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix: add default value for useSize |
| 🇨🇳 Chinese | - |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: packages/hooks/src/useSize/__tests__/index.test.ts [hunk 1]
```diff
@@ -16,11 +16,13 @@ jest.mock('resize-observer-polyfill', () => {
 describe('useSize', () => {
   it('with argument', () => {
     const hook = renderHook(() => useSize(document.body));
-    expect(hook.result.current).toEqual(undefined);
+    expect(hook.result.current).toEqual({ height: 0, width: 0 });
```
Comment (linhf123) [inline packages/hooks/src/useSize/__tests__/index.test.ts:19]: 如果传入的不是 `document.body` , 而是一个还没渲染的子元素，那么首次应该还是 `undefined` ?

File: packages/hooks/src/useSize/index.ts [hunk 2]
```diff
@@ -7,26 +7,23 @@ import useIsomorphicLayoutEffectWithTarget from '../utils/useIsomorphicLayoutEff
 type Size = { width: number; height: number };
 
 function useSize(target: BasicTarget): Size | undefined {
-  const [state, setState] = useRafState<Size>();
+  const el = getTargetElement(target);
+  const [state, setState] = useRafState<Size | undefined>(
```
Comment (linhf123) [inline packages/hooks/src/useSize/index.ts:11]: 这里会不会这样子会比较好? 因为在实践中使用，基本需要这么处理一下 `const w = size?.width || 0;`
` const [state, setState] = useRafState<Size>(
    { width: el.clientWidth || 0, height: el.clientHeight || 0 }
  );
`

File: packages/hooks/src/useSize/__tests__/index.test.ts (hunk 1 above)
Comment (li-jia-nan) [reply packages/hooks/src/useSize/__tests__/index.test.ts:19]: 是的，undefined 是预期的结果

File: packages/hooks/src/useSize/index.ts (hunk 2 above)
Comment (li-jia-nan) [reply packages/hooks/src/useSize/index.ts:11]: 没必要，这个会造成歧义，万一元素的 size 就是 0 呢

**Review by crazylxr — APPROVED** [review]

**Pull request merged**